tests/test_photoCalib.py

```diff
@@ -119,87 +138,106 @@ def _testPhotoCalibCenter(self, photoCalib, calibrationErr):
+        self.assertFloatsAlmostEqual(errFlux, result.error)
 
         # test that we get a correct magnitude err for the base instFlux
-        errMag = computeMagnitudeErr(self.instFluxErr, self.instFlux, calibrationErr, self.calibration, 1)
+        errMag = computeMagnitudeErr(self.instFluxErr,
+                                     self.instFlux,
+                                     calibrationErr,
+                                     self.calibration,
+                                     self.flux)
         result = photoCalib.instFluxToMagnitude(self.instFlux, self.instFluxErr)
-        self.assertEqual(0, result.value)
-        self.assertFloatsAlmostEqual(errMag, result.err)
+        self.assertEqual(self.mag, result.value)
+        self.assertFloatsAlmostEqual(errMag, result.error)
+        # and the same given an explicit point at the center
         result = photoCalib.instFluxToMagnitude(self.instFlux, self.instFluxErr, self.point0)
-        self.assertFloatsAlmostEqual(0, result.value)
-        self.assertFloatsAlmostEqual(errMag, result.err)
-
-        # test that we get a correct maggies err for base instFlux*1e-9
-        errFluxNano = computeMaggiesErr(self.instFluxErr, self.instFlux*1e-9,
-                                        calibrationErr, self.calibration, 1e-9)
-        result = photoCalib.instFluxToMaggies(self.instFlux*1e-9, self.instFluxErr)
-        self.assertFloatsAlmostEqual(1e-9, result.value)
-        self.assertFloatsAlmostEqual(errFluxNano, result.err)
-        result = photoCalib.instFluxToMaggies(self.instFlux*1e-9, self.instFluxErr, self.point0)
-        self.assertFloatsAlmostEqual(1e-9, result.value)
-        self.assertFloatsAlmostEqual(errFluxNano, result.err)
-
-        # test that we get a correct magnitude err for base instFlux*1e-9
-        errMagNano = computeMagnitudeErr(self.instFluxErr, self.instFlux*1e-9,
-                                         calibrationErr, self.calibration, 1e-9)
-        result = photoCalib.instFluxToMagnitude(self.instFlux*1e-9, self.instFluxErr)
-        self.assertFloatsAlmostEqual(22.5, result.value)
-        self.assertFloatsAlmostEqual(errMagNano, result.err)
-        result = photoCalib.instFluxToMagnitude(self.instFlux*1e-9, self.instFluxErr, self.point0)
-        self.assertFloatsAlmostEqual(22.5, result.value)
-        self.assertFloatsAlmostEqual(errMagNano, result.err)
+        self.assertFloatsAlmostEqual(self.mag, result.value)
+        self.assertFloatsAlmostEqual(errMag, result.error)
+
+        # test that we get a correct nJy err for flux2
+        errFlux2 = computeNanojanskyErr(self.instFluxErr,
+                                        self.instFlux2,
+                                        calibrationErr,
+                                        self.calibration,
+                                        self.flux2)
+        result = photoCalib.instFluxToNanojansky(self.instFlux2, self.instFluxErr)
```

Comment:
I worry this test is a bit circular ... it's comparing the version of the algorithm in the test file to that in the main code I guess?  Is there any verification that they are correct, or just match? (similar below)

Reply:
I'm not sure of another approach to this test: error calculations like this aren't in astropy, so I don't know of an external way to verify it.